bpf/lib/csum.h

```diff
@@ -62,7 +62,13 @@ static __always_inline int csum_l4_replace(struct __ctx_buff *ctx, int l4_off,
 					   struct csum_offset *csum,
 					   __be32 from, __be32 to, int flags)
 {
-	return l4_csum_replace(ctx, l4_off + csum->offset, from, to, flags | csum->flags);
+	__be32 sum = csum_diff(&from, sizeof(from), &to, sizeof(to), 0);
```

Comment:
Nit: You could maybe call `csum_diff_small()` here and avoid to type the two `sizeof()`. Not that it matters much.